iniparse/compat.py

```diff
@@ -185,14 +191,14 @@ def remove_section(self, section):
 
 
 class ConfigDict:
-    """Present a dict interface to a ini section."""
+    """Present a dict interface to an ini section."""
 
-    def __init__(self, cfg, section, vars):
-        self.cfg = cfg
-        self.section = section
+    def __init__(self, cfg: RawConfigParser, section: str, vars: "ConfigDict"):
```

Comment:
I wouldn't not change `vars` at this moment, because it probably does not cause any issues. Changing the API would cause more issues.

In theory we could create new API without `vars` and mark current API as deprecated using e.g. https://pypi.org/project/deprecation/ I'm not sure if it is worth of doing it.